src/prof.c

```diff
@@ -277,7 +277,8 @@ prof_sample_new_event_wait(tsd_t *tsd) {
 	 * otherwise bytes_until_sample would be 0 if u is exactly 1.0.
 	 */
 	uint64_t r = prng_lg_range_u64(tsd_prng_statep_get(tsd), 53);
-	double u = (r == 0U) ? 1.0 : (double)r * (1.0/9007199254740992.0L);
+	double u = (r == 0U) ? 1.0 : (double)((long double)r *
+	    (1.0L/9007199254740992.0L));
```

Comment:
Question: should we convert the constant to double instead? What's the warn msg?

Reply:
The warn message is mainly complaining about the implicit conversion between `double` and `long double`. I would say converting constant to either double or long double works as long as there is no concern on the precision.

Reply:
I'd say try making sure the precision remains the same, i.e. if it's doing an implicit conversion of r, then make it explicit.